### PR TITLE
hours but better than before

### PR DESCRIPTION
**Pull request by bandarussr**
# Description
Re-did the hours component to fit the new schema.. I did make some changes to the schema to make it work but this is causing issues in `src/server/actions/PdfGeneration.ts` which I do not know how to fix. (I've commented it out for now.. it also seems like the hours for services aren't being written to the pdf either..)

The new schema pretty much makes the hours of operation for the **agency** work the same as it would for a service. I've updated `src/utils/conversions.ts` and `src/utils/types/models.ts` so that it functions exactly like it does for services.

There is one more thing, the `Day[]` will contain time ranges that overlap, like Monday 2am-10pm and Monday 5pm-8pm. I'm not exactly sure where, but we need to collapse these ranges. I can't do this in the component, because it would be annoying for the user and the `useForm` keeps track of changes.

QoL changes: I added a scrollarea for the hours so the page height doesn't go crazy. I also added a shadow on the bottom that appears on hover and when there are enough elements to make it scrollable so you know that it is scrollable.

- #79 
- Previous pr: #92 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have assigned reviewers to this PR

### CODE REVIEW

File: src/app/[id]/page.tsx [hunk 1]
```diff
@@ -51,13 +52,14 @@ const steps = formSteps;
 
 export default function Form({ params }: { params: { id: string } }) {
   const [previousStep, setPreviousStep] = useState(-1);
-  const [currentStep, setCurrentStep] = useState(0);
+  const [currentStep, setCurrentStep] = useState(2);
```
Comment (cfisher36) [inline src/app/[id]/page.tsx:55]: needs to be reverted to step 0

File: src/components/Hours/index.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  } = useController({
+    name,
+    control,
+  });
+
+  const [hours, setHours] = useState((field.value as Hours[]) || []);
+
+  const [day, setDay] = useState<Day>('Monday');
+  const [open, setOpen] = useState<string>('12:00 AM');
+  const [close, setClose] = useState<string>('12:00 AM');
```
Comment (cfisher36) [inline src/components/Hours/index.tsx:33]: probably want close to default to 5pm

**Review by cfisher36 — CHANGES_REQUESTED** [review]
I'm getting a StrictModeError in the mongoErrorHandler when submitting that says hours is not in the schema. You may need to add the hours field to the mongoose schema.

Other than that, just a few small issues. Looks really clean

**Comment by cfisher36** [comment]
also, just merged main to add jacob's changes since there were some conflicts due to schema changes

**Comment by bandarussr** [comment]
Please check if I did the mongoose schema stuff correctly, I have no clue..

**Comment by cfisher36** [comment]
looks great, I changed it to auto switch to the next day on adding hours and fixed an issue where the open/close times were stuck on the default time. Could you look over those changes real quick to make sure it checks out with you?

**Review by cfisher36 — APPROVED** [review]

**Pull request merged**